src/readii/io/writers/plot_writer.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import dataclass, field
+from pathlib import Path
+
+import matplotlib
+
+from readii.io.writers.base_writer import BaseWriter
+from readii.utils import logger
+
+class PlotWriterError(Exception):
+    """Base exception for PlotWriter errors."""
+
+    pass
+
+
+class PlotWriterIOError(PlotWriterError):
+    """Raised when I/O operations fail."""
+
+    pass
+
+@dataclass
+class PlotWriter(BaseWriter):
+    """Class for managing file writing with customizable paths and filenames for plot figure files."""
+
+    overwrite: bool = field(
+        default=False,
+        metadata={
+            "help": "If True, allows overwriting existing files. If False, raises FileExistsError."
+        },
+    )
+
+    def save(self, plot:matplotlib.figure.Figure, **kwargs: str) -> Path:
+        """Save the plot to a .png file."""
+
+        logger.debug("Saving.", kwargs=kwargs)
+
+        # Generate the output path
+        out_path = self.resolve_path(**kwargs)
+
+        # Check if the output path already exists
+        if out_path.exists():
+            if not self.overwrite:
+                msg = f"File {out_path} already exists. \nSet {self.__class__.__name__}.overwrite to True to overwrite."
+                logger.exception(msg)
```

Comment:
the logger.exception is most useful for the main process that calls `writer.save`, as it might print out duplicated stack-traces